added getting started guide

iammuze opened the pull request:
Hi, @greenbreakfast `Getting Started Guide` has been added to issue #28 , please review it.

iammuze commented:
Hi @greenbreakfast, it appears that a check has failed in the pull request. Could you please take a look and address any issues?

greenbreakfast commented:
hi @iammuze, like I mentioned in my [last comment in issue 28](https://github.com/OnionIoT/documentation/issues/28#issuecomment-1930280861), this content should be in `docs/packages/intro.md` 

Please make the following changes:
* Move the written content to `docs/packages/intro.md` 
* Undo the changes to `sidebars.js` since the packages intro is already included
* In order to make the images more maintainable, move the images to an assets directory that's in the same directory as the article. 
  * *See the pin multiplexing article for details: 1) [how to include images in the article](https://github.com/OnionIoT/documentation/blob/main/docs/hardware-interfaces/pin-multiplexing.md?plain=1#L21), 2) [where the assets directory should be](https://github.com/OnionIoT/documentation/tree/main/docs/hardware-interfaces)*

greenbreakfast commented:
Yep that’s right On Feb 7, 2024, at 12:03 PM, Muze ***@***.***> wrote:﻿
Oh, I got confused. So, the content of this getting started guide will be reflect similar to the Device Tree Overlays chapter, right?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

iammuze commented:
> The build issue was happening because the packages branch didn't have the comment section implementation. I fixed that with a rebase. Added comments to the content - please take a look

Yeah, I also set up the `GiscusComment` package locally, but I didn't push it. I was thinking it might be a duplicate, but you resolved this issue. Thanks for that.

iammuze commented on docs/packages/intro.md:
```diff
@@ -2,18 +2,160 @@
+
+</details>
+
+## Packages Collection  
```
yeah, you're right.

iammuze commented on docs/packages/intro.md:
```diff
@@ -2,18 +2,160 @@
+
+The package Makefile acts as the central blueprint for the packages, specifying details such as the package name, version, compilation instructions needed for building the package, and additional instructions.
+
+:::tip
+For a detailed guide on Package Source Code, refer to [**OpenWRT**](https://openwrt.org/docs/guide-developer/source-code/start#the_openwrt_source_code) developer documentation.
+:::
```
alright, I'm commenting this for now and removing the OpenWRT reference. Later on, we can uncomment this `tip` to hyperlink our `docs` when we cover this topic in another iteration.

iammuze commented on docs/packages/intro.md:
```diff
@@ -2,18 +2,160 @@
+
+A Package Repository (Package Repo) is online storage for software packages that are made available for users to install on their devices using OPKG. Package repos contain collections of software packages.
+
+:::tip
+
+To further explore the package repo, refer to the official [**OnionIoT packages**](http://repo.onioniot.com/omega2/packages/openwrt-22.03.5/onion/) repo. This repository contains a variety of packages, including those for interacting with expansions, docks, and Omega2 device software applications.
+:::
```
yeah, I agree.

iammuze commented on docs/packages/intro.md:
```diff
@@ -2,18 +2,160 @@
 
-
-
 ## Articles in this section
 
 <DocCardList />
```
alright

iammuze commented on docs/packages/intro.md:
```diff
@@ -2,18 +2,160 @@
+
+<!-- Packages Diagram Source Code
+
+<iframe frameborder="0" style="width:100%;height:303px;" src="https://viewer.diagrams.net/?tags=%7B%7D&highlight=0000ff&edit=_blank&layers=1&nav=1#G1atYlLOH0eIhNPTF7XXZhrJ2OSuzK5ItA"></iframe>
```
Yes, I agree. However, commenting XML source code directly in the markdown file can make it appear cluttered, especially since it takes up 30 to 40 lines. One alternative approach could be to include the `.xml` file of the diagrams in the `assets` folder. We can then reference and comment out this file just after where we use the diagram in the markdown.

greenbreakfast replied:
That's a great idea! Let's do that!

iammuze commented on docs/packages/intro.md:
```diff
@@ -2,18 +2,160 @@
 title: OpenWRT Packages
 ---
 
+This getting started guide is a comprehensive cookbook that provides self-contained references for installing, creating, and configuring packages on Omega2 devices for practical applications. The guide includes a concise overview of key terms, such as packages, package manager, firmware, and package source code, offering a clear understanding of the essential concepts involved in working with Ome ...
```
Agreed, we can rephrase it as follows:

`This chapter provides an understanding on creating software packages for Omega2 devices. Furthermore, it also covers a brief explanation of the key terms that are essential to understand when dealing with OpenWRT packages.`

Please let me know if this works

greenbreakfast replied:
Reads really well. Since the chapter deals with packages and this article introduces the key terms, how about we spell that out:

This chapter provides an understanding on creating software packages for Omega2 devices. 
This article covers a brief explanation of the key terms that are essential to understand when dealing with OpenWRT packages.

greenbreakfast commented:
Looks good, merging these changes to main branch to publish the article
